app/models/concerns/system_like.rb

```diff
@@ -19,4 +19,24 @@ def compliant
     end
     result
   end
+
+  def total_rules(profile = nil)
+    return profile.results(self) if profile.present?
+    profiles.map do |p|
+      p.results(self)
+    end.flatten
+  end
+
+  def rules_passed(profile = nil)
+    total_rules(profile).count { |result| result }
```

Comment:
Maybe we should do some caching with `@rules_passed ||= ...` to reduce SQL queries (here and anywhere else we plan on calling a method multiple times).